pennylane/tape/tapes/jacobian_tape.py

```diff
@@ -546,3 +577,155 @@ def jacobian(self, device, params=None, **options):
             jac[:, i] = g
 
         return jac
+
+    def hessian(self, device, params=None, **options):
+        r"""Compute the Hessian of the parametrized quantum circuit recorded by the quantum tape.
+
+        The quantum tape can be interpreted as a simple :math:`\mathbb{R}^m \to \mathbb{R}^n` function,
+        mapping :math:`m` (trainable) gate parameters to :math:`n` measurement statistics,
+        such as expectation values or probabilities.
+
+        By default, the Hessian will be computed with respect to all parameters on the quantum tape.
+        This can be modified by setting the :attr:`~.trainable_params` attribute of the tape.
+
+        The Hessian can be currently computed using only the ``'analytic'`` method.
+
+        Args:
+            device (.Device, .QubitDevice): a PennyLane device
+                that can execute quantum operations and return measurement statistics
+            params (list[Any]): The quantum tape operation parameters. If not provided,
+                the current tape parameter values are used (via :meth:`~.get_parameters`).
+
+        Keyword Args:
+            method="analytic" (str): The differentiation method. Currently only
+                supports ``"analytic"``.
+            s1=pi/2 (float): the size of the shift for index i in the parameter-shift Hessian computations
+            s2=pi/2 (float): the size of the shift for index j in the parameter-shift Hessian computations
```

Comment:
Just an idea:
I'm assuming most use cases will have the same values for `s1` and `s2`.  
We could allow the user to either: 
* supply one number `s` that applies to both 
* supply `s1` and `s2` individually

Reply:
Alternatively, `s` could be a float _or_ a list!

```python
hessian(params, s=[pi/2, pi/4])
hessian(params, s=pi/2)
```

Although leaving it as two specific arguments is more explicit, and probably better for an internal method.

Reply:
Good idea! I do like the idea of adding the possibility to define both shifts through one parameter. Let me know if you guys want me to implement this, and which method, or if you want me to leave it as is for now.

Reply:
Thanks @soudy. No worries about implementing this idea now, the PR is already a great contribution as-is